Documentation: manually add IDs to all sections

szhorvat opened the pull request:
This fixes #1315

szhorvat commented:
Because of the force-pushes above, I'm just noting that this is ready to merge now. I verified that there are no more random section IDs, except in the license.

ntamas commented on doc/adjlist.xxml:
```diff
@@ -9,7 +9,7 @@
 <!-- doxrox-include about_adjlists -->
 
-<section><title>Adjacent vertices</title>
+<section id="adjacenct-vertices"><title>Adjacent vertices</title>
```
There is a typo here (`adjacenct` vs `adjacent`)

ntamas commented on doc/flows.xxml:
```diff
@@ -24,24 +24,24 @@
 </section>
 
-<section><title>Edge- and Vertex-Disjoint Paths</title>
+<section id="edge-and-vertexdisjoint-paths"><title>Edge- and Vertex-Disjoint Paths</title>
```
Please add a dash between `vertex` and `disjoint`

ntamas reviewed (changes requested):
I have spotted two typos (see comments), otherwise it's okay -- thanks a lot!

Feel free to merge when the typos are fixed.

szhorvat commented:
I ran a spell check on some, but not all IDs, and made some more corrections. The dashes went missing because part of the job was automated and my script forgot about `-` characters. I also removed the `igraph-` prefixes in `error.xxml` and added in some dashes.

Pushed separate commit so you can see the changes. We can then squash-and-merge.

ntamas commented:
Thanks!
